ssz_serialization/codec.nim

```diff
@@ -19,18 +19,20 @@ import
 export
   types
 
-template raiseIncorrectSize*(T: type) =
+func raiseMalformedSszError*(T: type, msg: string) {.raisesssz, noinline, noreturn.} =
```

Comment:
There will be less bloat if everything is passed through a single non-generic function that takes the type name as a parameter. In other words, `raiseMalformedSszError` can remain a template that routes the call to a new non-generic function.